pkg/gateway/operations/putobject.go

```diff
@@ -74,11 +73,22 @@ func handleCopy(w http.ResponseWriter, req *http.Request, o *PathOperation, copy
 		return // operation already failed
 	}
 	ent.CreationDate = time.Now()
-	ent.Path = o.Path
-	err := o.Catalog.CreateEntry(req.Context(), o.Repository.Name, o.Reference, *ent)
+	address, err := o.Catalog.GenerateName()
 	if err != nil {
-		o.Log(req).WithError(err).Error("could not write copy destination")
-		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInvalidCopyDest))
+		_ = o.EncodeError(w, req, errors.Codes.ToAPIErr(errors.ErrInternalError))
```

Comment:
Can we have a more detailed error than "internal error"?  Also we will need some way to debug this, and we either have no log or we lost some detail in the log.